web/search.dart

```diff
@@ -203,24 +211,56 @@ void initializeSearch(
         query, "<strong class='tt-highlight'>$sanitizedText</strong>");
   }
 
+  var categoriesMap = <String, Element>{};
+
+  // Maps a suggestions library/class to the other suggestions that search found if any.
+  void categorize(Element suggestionLibrary, Element suggestion) {
+    var input = suggestionLibrary.innerHtml;
+
+    if (input == null) {
+      return;
+    }
+
+    var element = categoriesMap[input];
+    if (element != null) {
+      element.append(suggestion);
+      categoriesMap.update(input, (value) => element);
+    } else {
+      suggestionLibrary.append(suggestion);
+      categoriesMap[input] = suggestionLibrary;
+    }
+  }
+
+  // Using the name of the library/class creates the Element for it.
+  Element createCategory(String lib, String href) {
+    var categoryTitle = document.createElement('a');
+    categoryTitle.setAttribute('href', href);
+    categoryTitle.classes.add('tt-category-title');
+    categoryTitle.innerHtml = lib;
+    return categoryTitle;
+  }
+
   Element createSuggestion(String query, IndexItem match) {
     var suggestion = document.createElement('div');
     suggestion.setAttribute('data-href', match.href ?? '');
     suggestion.classes.add('tt-suggestion');
 
-    var suggestionTitle = document.createElement('span');
+    var suggestionTitle = document.createElement('div');
     suggestionTitle.classes.add('tt-suggestion-title');
     suggestionTitle.innerHtml =
         highlight('${match.name} ${match.type.toLowerCase()}', query);
-
     suggestion.append(suggestionTitle);
 
-    if (match.enclosedBy != null) {
-      var fromLib = document.createElement('div');
-      fromLib.classes.add('search-from-lib');
-      fromLib.innerHtml = 'from ${highlight(match.enclosedBy!.name, query)}';
-
-      suggestion.append(fromLib);
+    // The new one line description to use in the search suggestions.
+    if (match.desc != '') {
+      var inputDesc = document.createElement('div');
+      inputDesc.classes.add('one-line-description');
+      var innerHtml = '';
+      if (match.desc != null) {
```

Comment:
Some of this is a little weird in the latest commit. We check on line 255 that `match.desc != ''`, then check here that it's not `null`. It seems like it should just be one check?

Reply:
You are right, it should just be checked once. New commit addresses this.